core/templates/components/state-directives/rule-editor/rule-type-selector.directive.ts

```diff
@@ -46,25 +46,16 @@ angular.module('oppia').directive('ruleTypeSelector', [function() {
 
           var ruleTypesToDescriptions = INTERACTION_SPECS[
             StateInteractionIdService.savedMemento].rule_descriptions;
+          // TODO(bhenning): The order of choices should be meaningful. E.g.,
+          // having "is equal to" for most interactions first makes sense. They
+          // should ideally be ordered based on likelihood of being used.
           for (var ruleType in ruleTypesToDescriptions) {
             choices.push({
               id: ruleType,
               text: $filter('replaceInputsWithEllipses')(
                 ruleTypesToDescriptions[ruleType])
             });
           }
-          // TODO(bhenning): The order of choices should be meaningful. E.g.,
-          // having "is equal to" for most interactions first makes sense. They
-          // should ideally be ordered based on likelihood of being used.
-          choices.sort(function(a, b) {
```

Comment:
I removed this since alphabetical order didn't make sense for a lot of rules. Instead, it now follows the ordering present in interaction_specs so it gives more control per interaction.

Reply:
The order in interaction_specs is a dict, right? So actually that would mean that the order is undefined?

If so, what I would suggest is: "Equals", followed by the rest in alphabetical order (so that it's at least fully defined in all browsers).

(If you decide to leave it as-is, though, please leave a comment to explain.)

Reply:
Hmm, yeah, it is a dict, but I checked the exploration editor a few times and the order was preserved, as was defined in interaction_specs. Maybe the order of keys stays the same in the dict once defined?
Added a comment.

Reply:
Ah -- it is undefined, but some browser implementations just follow the existing order. This is not guaranteed, though, and is entirely subject to change, so we should not be relying on this behaviour.

Reply:
Done